packages/manager/apps/pci-kubernetes/src/api/hooks/usePlanToRegionAvailability.ts

```diff
@@ -0,0 +1,70 @@
+import { useMemo } from 'react';
+
+import { TProductAvailability, useProductAvailability } from '@ovh-ux/manager-pci-common';
+
+import useStandardPlanAvailable from '@/hooks/useStandardPlanAvailable';
+import { TClusterPlanEnum } from '@/types';
+
+type TRegion = {
+  name: string;
+  datacenter: string;
+  continentCode: string;
+  enabled: boolean;
+  type: string;
+  availabilityZones: string[];
+  codes: string[];
+};
+
+export const transformToRegions = (
+  data?: TProductAvailability,
+  standardIsActivated = true,
+): TRegion[] =>
+  Object.values(
+    data?.plans
+      ?.flatMap((plan) => plan.regions.map((region) => ({ ...region, planCode: plan.code })))
+      .filter(({ planCode }) =>
+        standardIsActivated
+          ? true
+          : planCode.includes(TClusterPlanEnum.FREE) || planCode.includes('.3az'),
+      )
+      .reduce<Record<string, TRegion>>(
+        (acc, { planCode, ...region }) => ({
+          ...acc,
+          [region.name]: {
+            ...region,
+            codes: [...(acc[region.name]?.codes ?? []), planCode],
+          },
```

Comment:
Question: Can there be multiple region with the same name ? If yes, you will override everything execept codes in [region.name] with la last region with that name, is that wanted ?